recipes-devtools/rust/rust.inc

```diff
@@ -315,6 +366,10 @@ def rust_gen_mk_cfg(d, thing):
         '-e', '/^CFG_GCCISH_CXXFLAGS/ s;$; {};'.format(cxxflags_for(d, thing)),
         '-e', '/^CFG_GCCISH_LINK_FLAGS/ s;$; {};'.format(ldflags),
 
+        # The mipsel target specify flags for the mips32. Replace them with
+        # mips32r2 if that's what we use
+        '-e', 's/mips32/mips32r2/g' if is_mips32r2 else '',
```

Comment:
I agree it is not very nice.

The flags from yocto are fine, they specifiy the cpu (e.g. -march=34kc). The problem is the mk/cfg/ files specifies -mips32 (equivalent to -march=mips32) for CFG_JEMALLOC_CFLAGS, CFG_GCCISH_CFLAGS, CFG_GCCISH_LINK_FLAGS. But the -march=34kc flag is placed after the -mips32, so perhaps that will always be chosen. In that case this substitution is unnecessary. Do you know if that is the case?

Here's an example of what I mean:
`CFG_GCCISH_CFLAGS_mipsel-poky-linux := -Wall -g -fPIC -mips32 -mabi=32  -mel -mabi=32 -msoft-float -march=34kc -msynci --sysroot=...`

However these files are completely gone by now so perhaps it would be better to put effort into bumping to rust 1.15